### PR TITLE
Feat : Save and import side settings

### LINKED ISSUE #78: Remember settings on tab sleep/crash
When the user changes settings for a codec, these should be saved somewhere. When they select that codec later, those settings should be restored.

Because of this, we probably want a "reset" button.

Settings should be remembered for each side of the screen. So side * codec.

The codec selected on each side should be remembered between page loads too.

### LINKED ISSUE #430: User Presets
Would be nice to have user customisable and savable presets so that I can quickly pick between my favorite settings, allowing faster batch editing. Presumably, this would also need a way to title, edit and delete presets. Similar to #78.

P.S. Keep up the good work! 👍

### LINKED ISSUE #1268: Please save settings on website
**Is your feature request related to a problem? Please describe.**
I need to process a set of images (resize, turn to webp, change quality). I have to set these settings for every image I upload.

**Describe the solution you'd like**
Save last selected options to localStorage.

### LINKED ISSUE #1299: 2 ideas: 4-way compare, copy&paste the settings
**Is your feature request related to a problem? Please describe.**
2-way compare is sometimes not enough, especially given the amount of parameters to tune.

**Describe the solution you'd like**
2-way compare is sometimes not enough. It can be easily extended into 4-way compare, by having a horizontal and a vertical line.

Also, a way to copy&paste settings (JSON) would be a very useful feature. It would allow: (a) to easily replicate settings for comparison, (b) save settings for later use, (b) share the settings between people.

**Does other service/app have this feature?**
No.

### PR DESCRIPTION
**Pull request by harsh26shah03**
There were requests from multiple users that
they use squoosh for compression but for each
iteration side settings resets to default
causing issues and there is no way to save and
import side settings.

There will be two buttons adjacent to copy-over

save side settings : This will save side encoder
and latest settings to localstorage of browser

import side settings : This will import side encoder and latest settings from localstorage of browser and replace the existing settings

Also if there are saved settings in locaStorage then whenever user loads the app it will take that settings and populate the side so user do not have to repeatedly enter same settings for similar compression operation subject to user has saved side settings

Update:1

Import settings button remains disabled if there
is nothing to import

Whenever the side setting is saved there will be
event fired and eventually listened to enable import button

All 2 operations show notifications now

Import notification has undo option now

https://user-images.githubusercontent.com/48324810/229767370-c4749545-6190-4f13-8203-757477ea08a1.mov

Closes #1299 
Closes #1268 
Closes #78
Closes #430

### CODE REVIEW

**Comment by harsh26shah03** [comment]
@argyleink have a look into this new feature

**Comment by surma** [comment]
Love this. I could see this being super useful. Only feedback at a skim is that the icons need some love, but I am sure @argyleink has ideas on that front, too :)

File: src/client/lazy-app/icons/index.tsx [hunk 1]
```diff
@@ -97,3 +97,27 @@ export const SwapIcon = () => (
...
+      data-name="layer2"
+      d="M41.9 30a16 16 0 1 0 16 16 16 16 0 0 0-16-16zm7.4 15.4a2 2 0 0 1-2.8 0l-2.6-2.6V54a2 2 0 0 1-4 0V42.8l-2.6 2.6a2 2 0 1 1-2.8-2.8l7.4-7.4 7.4 7.4a2 2 0 0 1 0 2.8z"
+    ></path>
+    <ellipse data-name="layer1" cx="29.1" cy="10" rx="23" ry="8"></ellipse>
+    <path
+      data-name="layer1"
+      d="M29.1 29h2.6a19.7 19.7 0 0 1 14.7-2.4c3.6-1.2 5.6-2.5 5.6-3.5v-6.3C47 20.4 38 22 29 22s-18-1.7-23-5.2V23c.1 2 9.6 6 23.1 6zm-7 17c0-.5 0-.9.1-1.4-6.9-.7-13-2.6-16.1-5.4V46c0 3.7 7.5 6.9 17.6 7.8a19.9 19.9 0 0 1-1.6-7.8zm.8-5.3a19.9 19.9 0 0 1 4.1-7.8c-8.7-.3-17.2-2.2-20.9-5.8V35c0 1.7 6.7 4.7 16.8 5.7z"
+    ></path>
+  </svg>
+);
```
Comment (argyleink) [inline src/client/lazy-app/icons/index.tsx]:
```suggestion
export const SaveIcon = () => (
  <svg viewBox="0 0 24 24">
    <g fill="none" stroke-linecap="round" stroke-linejoin="round" stroke-width="2">
      <path d="M12.501 20.93c-.866.25-1.914-.166-2.176-1.247a1.724 1.724 0 0 0-2.573-1.066c-1.543.94-3.31-.826-2.37-2.37a1.724 1.724 0 0 0-1.065-2.572c-1.756-.426-1.756-2.924 0-3.35a1.724 1.724 0 0 0 1.066-2.573c-.94-1.543.826-3.31 2.37-2.37c1 .608 2.296.07 2.572-1.065c.426-1.756 2.924-1.756 3.35 0a1.724 1.724 0 0 0 2.573 1.066c1.543-.94 3.31.826 2.37 2.37a1.724 1.724 0 0 0 1.065 2.572c1.074.26 1.49 1.296 1.252 2.158M19 22v-6m3 3l-3-3l-3 3"/><path d="M9 12a3 3 0 1 0 6 0a3 3 0 0 0-6 0"/>
    </g>
  </svg>
);
```

File: src/client/lazy-app/icons/index.tsx [hunk 2]
```diff
@@ -97,3 +97,27 @@ export const SwapIcon = () => (
...
+
+export const ImportIcon = () => (
+  <svg viewBox="0 0 64 64" xmlns="http://www.w3.org/2000/svg" role="img">
+    <path
+      d="M49.189 31.052l-4.378-4.104L35 37.414V2h-6v35.066l-9.751-11.05-4.498 3.969 17.182 19.473 17.256-18.406z"
+      data-name="layer2"
+    ></path>
+    <path d="M52 45v11H12V45H6v17h52V45h-6z" data-name="layer1"></path>
+  </svg>
+);
```
Comment (argyleink) [inline src/client/lazy-app/icons/index.tsx]:
```suggestion
export const ImportIcon = () => (
  <svg viewBox="0 0 24 24">
    <g fill="none" stroke-linecap="round" stroke-linejoin="round" stroke-width="2">
      <path d="M12.52 20.924c-.87.262-1.93-.152-2.195-1.241a1.724 1.724 0 0 0-2.573-1.066c-1.543.94-3.31-.826-2.37-2.37a1.724 1.724 0 0 0-1.065-2.572c-1.756-.426-1.756-2.924 0-3.35a1.724 1.724 0 0 0 1.066-2.573c-.94-1.543.826-3.31 2.37-2.37c1 .608 2.296.07 2.572-1.065c.426-1.756 2.924-1.756 3.35 0a1.724 1.724 0 0 0 2.573 1.066c1.543-.94 3.31.826 2.37 2.37a1.724 1.724 0 0 0 1.065 2.572c1.088.264 1.502 1.323 1.242 2.192M19 16v6m3-3l-3 3l-3-3"/><path d="M9 12a3 3 0 1 0 6 0a3 3 0 0 0-6 0"/>
    </g>
  </svg>
);
```

File: src/client/lazy-app/Compress/Options/style.css [hunk 3]
```diff
@@ -119,3 +119,20 @@
     fill: var(--header-text-color);
   }
 }
+
+.save-button,
+.import-button {
+  composes: title-button;
+
+  svg {
+    fill: var(--header-text-color);
```
Comment (argyleink) [inline src/client/lazy-app/Compress/Options/style.css]:
```suggestion
    stroke: var(--header-text-color);
```

**Review by argyleink — CHANGES_REQUESTED** [review]
I thought these icons communicate the intent a little better? thoughts?

Screenshot 2023-04-04 at 9 46 33 AM

**Comment by harsh26shah03** [comment]
@argyleink changes are in now, your suggestion looks 👏

**Review by argyleink — APPROVED** [review]
lgtm

**Pull request merged**